garbage-collection/gc_cleanup.py

```diff
@@ -0,0 +1,148 @@
+        # Get specific metadata based on workflow type
+        if args.adhoc:
+            exists = False
+            for word in args.starts_with + args.label_selector:
+                if key.startswith(word):
+                    #skip
+                    exists = True
+                elif word in workflow['metadata']['labels']:
+                    #skip
+                    exists = True
+            if not exists and int(time_since_completion) > int(args.max_age_hrs):
+                workflows_expired.append(key)
+                # print("deleting workflow pod {} that completed {} hrs ago".format(key,time_since_completion))
+                pods = get_pods(workflow,args.namespace)
+                for pod in pods:
+                    if not args.dry_run:
+                        delete_pods(pod,args.namespace,body)
+                    else:
+                        logging.info("dry_run flag set, would have deleted {}".format(pod))
+            else:
+                workflows_not_expired.append(key)
+        else:
+            exists = False
+            for word in args.starts_with + args.label_selector:
+                if key.startswith(word):
+                    #skip
+                    exists = True
+                elif word in workflow['metadata']['labels']:
+                    #skip
+                    exists = True
+            if exists and int(time_since_completion) > int(args.max_age_hrs):
+                workflows_expired.append(key)
+                # print("deleting workflow pod {} that completed {} hrs ago".format(key,time_since_completion))
+                pods = get_pods(workflow, args.namespace)
+                for pod in pods:
+                    if not args.dry_run:
+                        delete_pods(pod,args.namespace,body)
+                    else:
+                        logging.info("dry_run flag set, would have deleted {}".format(pod))
+            else:
+                workflows_not_expired.append(key)
+    logging.info("expired workflows: {}".format(workflows_expired))
+
+
+def main():
+  logging.basicConfig(level=logging.INFO)
+  logging.getLogger().setLevel(logging.INFO)
+
+# initiate the parser
+parser = argparse.ArgumentParser(description = 'a garbage collection utility for cleaning up argo workflows')
+
+# add in a namespace argument
+parser.add_argument("-n", "--namespace", default="default", type=str, help=("The custom resource's namespace."))
+
+# add in a group argument
+parser.add_argument("-grp", "--group", default="argoproj.io", type=str, help=("The custom resource's group name."))
+
+# add in a version argument
+parser.add_argument("-version", default="v1alpha1", type=str, help=("The custom resource's version."))
+
+# add in a plurals argument
+parser.add_argument("-p", "--plural", default="workflows", type=str, help=("The custom resource's plural name to filter by."))
+
+# add in a plurals argument
```

Comment:
this is an incorrect comment. Also, comments like this don't really add much information